loopin.go

```diff
@@ -209,6 +247,63 @@ func newLoopInSwap(globalCtx context.Context, cfg *swapConfig,
 	}, nil
 }
 
+// awaitProbe waits for a probe payment to arrive and cancels it. This is a
+// workaround for the current lack of multi-path probing.
+func awaitProbe(ctx context.Context, lnd lndclient.LndServices,
+	probeHash lntypes.Hash) (chan error, error) {
+
+	// Subscribe to the probe invoice.
+	updateChan, errChan, err := lnd.Invoices.SubscribeSingleInvoice(
+		ctx, probeHash,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait in the background for the probe to arrive.
+	probeResult := make(chan error, 1)
+
+	go func() {
+		for {
+			select {
+			case update := <-updateChan:
+				switch update.State {
+				case channeldb.ContractAccepted:
+					log.Infof("Server probe successful")
+					probeResult <- nil
+
+					// Cancel probe invoice so that the
+					// server will know that its probe was
+					// successful.
+					err := lnd.Invoices.CancelInvoice(
+						ctx, probeHash,
+					)
+					if err != nil {
+						log.Errorf("Cancel probe invoice: %v", err)
+					}
+
+					return
+
+				case channeldb.ContractSettled:
+					log.Errorf("Impossible that probe " +
```

Comment:
Even though impossible, will this hang the client, since we never write to `probeResult`? Also, should we panic here instead?

Reply:
I don't think this hangs the client, because after the probe init call returns from the server, we cancel the context and exit this goroutine.

Reply:
`panic` would bring down the whole server. Wouldn't want to risk that if this would happen because of a bug somewhere, especially because the probe result isn't critical.

Reply:
For consistency I changed it to also write to `probeResult` when the invoice is settled

Reply:
> `panic` would bring down the whole server. Wouldn't want to risk that if this would happen because of a bug somewhere, especially because the probe result isn't critical.

sgtm